### PR TITLE
GHC-111: Add laboratory tests

### PR DESCRIPTION
**Pull request by basrem**

### CODE REVIEW

File: configuration/concepts/commons.csv [hunk 1]
```diff
@@ -2,4 +2,8 @@ Uuid,Void/Retire,Same as mappings,Fully specified name:en,Short name:en,Fully sp
 67218dc9-aee7-40d1-ba78-d29dd2731333,,CIEL:1362,One plus,+,Uno más,+,,Misc,N/A,,,,,,,,,,,
 55800eb1-4e0f-4a75-a12b-aa1e10c64e81,,CIEL:1363,Two plus,++,Dos más,++,,Misc,N/A,,,,,,,,,,,
 17327c32-0a60-4f76-85b5-24379fc72ad5,,CIEL:1364,Three plus,+++,Tres más,+++,,Misc,N/A,,,,,,,,,,,
-f0853fb9-5541-4e13-b79a-e3b438a629cb,,CIEL:1365,Four plus,++++,Cuatro más,++++,,Misc,N/A,,,,,,,,,,,
\ No newline at end of file
+f0853fb9-5541-4e13-b79a-e3b438a629cb,,CIEL:1365,Four plus,++++,Cuatro más,++++,,Misc,N/A,,,,,,,,,,,
+ff07388d-9309-4bc2-a9e9-e31031352d24,,,A,,A,,,Misc,N/A,,,,,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/commons.csv]: Add short names as well for these

File: configuration/concepts/lab_tests.csv [hunk 2]
```diff
@@ -95,14 +100,56 @@ d3966267-5882-437b-ae14-18dee9657d14,,,RBCs,Hematíes,Hematíes,RBCs,,LabTest,Co
...
 14e99cf4-d294-4905-8837-40e704d8a6ad,,,Vaginal Candidiasis,Candidiasis vaginal,Candidiasis vaginal,Vaginal Candidiasis,,LabTest,Text,,,,,,,,,,
-ab6588f6-87ee-45ca-a6c5-dc1337f83469,,,Gonorrhea,Gonorrea,Gonorrea,Gonorrhea,,LabTest,Coded,,Positive; Negative;,,,,,,,,
-637b4f48-8717-45e2-a5f8-96869e3bb768,,,Chlamydia,Clamidia,Clamidia,Chlamydia,,LabTest,Coded,,Positive; Negative;,,,,,,,,
+ab6588f6-87ee-45ca-a6c5-dc1337f83469,,,Gonorrhea,Gonorrea,Gonorrea,Gonorrhea,,LabTest,Coded,,Positive; Negative,,,,,,,,
+637b4f48-8717-45e2-a5f8-96869e3bb768,,,Chlamydia,Clamidia,Clamidia,Chlamydia,,LabTest,Coded,,Positive; Negative,,,,,,,,
+ff74d247-aa76-4f2c-bcfd-01be27559681,,,ABO,ABO,ABO,ABO,,LabTest,Coded,,A;B;AB;O,,,,,,,,
+9b51c857-2d8e-4ef9-bece-a80cbd175ae6,,,Rhesus Factor,factor rhesus,RH,RH,,LabTest,Coded,,Positive; Negative,,,,,,,,
+d3b33cb4-4f6a-44e3-abd5-82e794d17ef7,,,Blood Grouping,Agrupación Sanguínea,Agrupación Sanguínea,Blood Grouping,,LabSet,N/A,,,ABO;Rhesus Factor,,,,,,,
+a5bee8bd-530a-4d5f-bf34-27f55b3b2515,,,Major,Principal,Principal,Major,,LabTest,Text,,,,,,,,,,
+cf971905-bd1e-4901-8633-a77f5263a71c,,,Minor,Menor,Menor,Minor,,LabTest,Text,,,,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: We change these to **Major Crossmatch** and **Minor Crossmatch**

File: configuration/concepts/lab_tests.csv [hunk 3]
```diff
@@ -95,14 +100,56 @@ d3966267-5882-437b-ae14-18dee9657d14,,,RBCs,Hematíes,Hematíes,RBCs,,LabTest,Co
...
+0733fbb6-41fa-457b-a57f-d6653caa85f8,,,Helminths,Helmintos,Helmintos,Helminths,,LabSet,N/A,,,Ascaris lumbricoides;Trichuris trichiura;Hymenolepis nana;Taenia sp,,,,,,,
+87b1df6c-2cb3-4e76-bb6a-d89045f35f07,,,Wright Test,Prueba de Wright,Prueba de Wright,Wright Test,,LabTest,Text,,,,,,,,,,
+64bddc56-09c6-4eb7-8e31-baabc15c0f50,,,Crystals,Cristales,Cristales,Crystals,,LabTest,Text,,,,,,,,,,
+d8d5293e-9289-484a-b969-05659c076e38,,,Cast,emitir,emitir,Cast,,LabTest,Text,,,,,,,,,,
+8a61c5ee-a83c-4b48-a30b-a432c6e2409e,,,Yeast Cells,Células de levadura,Células de levadura,Yeast Cells,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+11eb2dd2-fa80-4560-bd5d-bffe2d0ce064,,,Mucus,Moco,Moco,Mucus,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+cf1cff9a-0df8-4e2d-8abf-f96a3392f8f0,,,Red Blood Cell Count,Recuento de glóbulos rojos,Recuento de glóbulos rojos,Red Blood Cell Count,,LabTest,Numeric,,,,,cells per field,,,,,
+ef036fb3-6e8c-4c79-9055-54913e4ef70f,,,White Blood Cell Count,Recuento de glóbulos blancos,Recuento de glóbulos blancos,White Blood Cell Count,,LabTest,Numeric,,,,,cells per field,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: We change this to **Red Blood Cells in urine** and **White Blood Cells in urine**

File: configuration/concepts/lab_tests.csv [hunk 4]
```diff
@@ -95,14 +100,56 @@ d3966267-5882-437b-ae14-18dee9657d14,,,RBCs,Hematíes,Hematíes,RBCs,,LabTest,Co
...
+9a89b6e4-143b-4c03-b929-b085621f75e5,,,Epithelial Tissue,Tejido epitelial,Tejido epitelial,Epithelial Tissue,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+3fed1729-e4e3-434c-946c-58b3f9e8d963,,,Bacterial Cells,Células bacterianas,Células bacterianas,Bacterial Cells,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+666bee99-7881-418d-8ea1-5d2301aa0640,,,Microscopy Set,Juego de microscopía,Juego de microscopía,Microscopy Set,,LabSet,N/A,,,Crystals;Cast;Yeast Cells;Mucus;Red Blood Cell Count;White Blood Cell Count;Epithelial Tissue;Other Observations,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: Let's introduce a test called **Other Test Observations** instead of **Other Observations**

File: configuration/concepts/lab_tests.csv [hunk 5]
```diff
@@ -61,7 +67,6 @@ c288ec60-4755-4f42-a4de-e197bf9ba0fa,,,Antistreptolysin O,Antiestreptolisina O,A
 00df2515-b7c2-45bb-ae5e-8872e9f2446f,,,Amylase,Amilasa,Amilasa,Amylase,,LabTest,Numeric,,,,,,,,,,
 6b12c39e-dfa8-4cdf-aa05-ca7e577d2fd2,,,Lipase,Lipasa,Lipasa,Lipase,,LabTest,Numeric,,,,,,,,,,
 3c42b57e-2b96-48d1-8f54-eb325d268882,,,Rheumatoid Factor,Factor reumatoide,Factor reumatoide,Rheumatoid Factor,,LabTest,Coded,,Positive;Negative,,,,,,,,
-c288ec60-4755-4f42-a4de-e197bf9ba0fa,,,Antistreptolysin O,Antiestreptolisina O,Antiestreptolisina O,Antistreptolysin O,,LabTest,Coded,,Positive;Negative,,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv:64]: Why did you delete this?

**Review by reagan-meant — CHANGES_REQUESTED** [review]
Add all **serology** tests under Blood sample

File: configuration/concepts/lab_tests.csv (hunk 3 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv]: Done

File: configuration/concepts/lab_tests.csv (hunk 2 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv]: Done

File: configuration/concepts/lab_tests.csv (hunk 4 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv]: Done

File: configuration/concepts/lab_tests.csv (hunk 5 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv:64]: line 60 was duplicated at line 64 which was deleted

File: configuration/concepts/commons.csv (hunk 1 above)
Comment (basrem) [reply configuration/concepts/commons.csv]: Added these blood groups to labtest csv

File: configuration/concepts/lab_tests.csv [hunk 6]
```diff
@@ -1,110 +1,171 @@
...
+11eb2dd2-fa80-4560-bd5d-bffe2d0ce064,,,Mucus,Moco,Moco,Mucus,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+cf1cff9a-0df8-4e2d-8abf-f96a3392f8f0,,,Red Blood Cells in Urine,Glóbulos rojos en la orina,Glóbulos rojos en la orina,Red Blood Cells in Urine,,LabTest,Numeric,,,,,cells per field,,,,,
+ef036fb3-6e8c-4c79-9055-54913e4ef70f,,,White Blood Cells in Urine,Glóbulos blancos en la orina,Glóbulos blancos en la orina,White Blood Cells in Urine,,LabTest,Numeric,,,,,cells per field,,,,,
+9a89b6e4-143b-4c03-b929-b085621f75e5,,,Epithelial Tissue,Tejido epitelial,Tejido epitelial,Epithelial Tissue,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+3fed1729-e4e3-434c-946c-58b3f9e8d963,,,Bacterial Cells,Células bacterianas,Células bacterianas,Bacterial Cells,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+57bf229a-b469-49f1-9d29-3f6e90120b8d,,,"Other Test Observations ",Otras observaciones de prueba,Otras observaciones de prueba,"Other Test Observations ",,LabSet,Text,,,,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: Let's remove quotation marks

File: configuration/concepts/lab_tests.csv [hunk 7]
```diff
@@ -1,110 +1,171 @@
...
+2cbfd68c-b45a-4c25-b9a3-650169d87940,,,Blood,Sangre,Sangre,Blood,,Sample,N/A,,,CBC; Lipid Profile; Coagulation Profile; Thyroid Functional Test; Renal Functional Test; Liver Functional Test; Rapid malaria; Rapid HIV; Blood culture; Fasting blood glucose; Random blood glucose; HbA1C; mALB; 1e3e425c-eee2-4d61-8652-dd20239ec159; Hematocrit; Red Blood Cells; White Blood Cells; Platelets; Plateletcrit;Mean Platelet Volume;PDWs;PDWc;P-LCC;P-LCR; MCV; MCH; MCHC; RDWs; RDWc; Reticulocytes; Neutrophil; Lymphocyte; Monocyte; Basophil; Eosinophil; Creatinine; BUN; Urea; Sodium; Potassium; Calcium Test; Bicarbonate; Chloride; Phosphate; Serum magnesium; Uric acid; Total Protein; Total Bilirubin; Direct Bilirubin; albumin; ALT; AST; ALP; Total Cholesterol; HDL; LDL; Triglycerides; PT; aPTT; INR; C-Reactive Protein; Erythrocyte Sedimentation Rate; TSH; T3; T4; Amylase; Lipase; Rheumatoid Factor; Antistreptolysin O; Prolactin; Sex Hormones;Chlamydia;Gonorrhea;Hepatitis B;Hepatitis c;Helicobacter pylori Antigen;Antistreptolysin O;Rapid HIV;C-Reactive Protein; Rapid Plasma Reagin;Pregnancy;Rheumatoid Arthritis,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: Change to **Hepatitis C** instead of **Hepatitis c**

**Review by reagan-meant — CHANGES_REQUESTED** [review]

File: configuration/concepts/lab_tests.csv [hunk 8]
```diff
@@ -95,27 +100,75 @@ d3966267-5882-437b-ae14-18dee9657d14,,,RBCs,Hematíes,Hematíes,RBCs,,LabTest,Co
...
-74ab27c4-c745-45ff-b7e6-e8dbaf933458,,,Trichomonas,Trichomonas,Trichomonas,Trichomonas,,LabTest,Coded,,Positive; Negative;,,,,,,,,
+4bb7df94-7b92-4e5f-8221-064a76d65e17,,,Yeast,Hongos,Hongos,Yeast,,LabTest,Coded,,Positive; Negative,,,,,,,,
+b2a75309-4eb5-414d-9f29-8ecb4ac7dd32,,,Bacterial vaginosis,Vaginosis bacteriana,Vaginosis bacteriana,Bacterial vaginosis,,LabTest,Coded,,Positive; Negative,,,,,,,,
+74ab27c4-c745-45ff-b7e6-e8dbaf933458,,,Trichomonas,Trichomonas,Trichomonas,Trichomonas,,LabTest,Coded,,Positive; Negative,,,,,,,,
 14e99cf4-d294-4905-8837-40e704d8a6ad,,,Vaginal Candidiasis,Candidiasis vaginal,Candidiasis vaginal,Vaginal Candidiasis,,LabTest,Text,,,,,,,,,,
-ab6588f6-87ee-45ca-a6c5-dc1337f83469,,,Gonorrhea,Gonorrea,Gonorrea,Gonorrhea,,LabTest,Coded,,Positive; Negative;,,,,,,,,
-637b4f48-8717-45e2-a5f8-96869e3bb768,,,Chlamydia,Clamidia,Clamidia,Chlamydia,,LabTest,Coded,,Positive; Negative;,,,,,,,,
+0d3ddc36-7670-4303-8aaf-5f755afe7e99,,,Rheumatoid Arthritis,Artritis Reumatoide,Artritis Reumatoide,Rheumatoid Arthritis,,LabTest,Coded,,Positive; Negative,,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: @basrem Is this also a test given? Seen line 69

File: configuration/concepts/lab_tests.csv [hunk 9]
```diff
@@ -95,27 +100,75 @@ d3966267-5882-437b-ae14-18dee9657d14,,,RBCs,Hematíes,Hematíes,RBCs,,LabTest,Co
...
+088253e3-b556-4e82-b646-2bb784a46577,,,Pregnancy,El embarazo,El embarazo,Pregnancy,,LabTest,Coded,,Positive; Negative,,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: @basrem We change this concept to 'Blood HCG (Pregnancy)'

**Review by reagan-meant — COMMENTED** [review]
Lets add the following
Prostate Specific Antigen
Bacteria in Microscopy section
Color 
Aspect

File: configuration/concepts/lab_tests.csv (hunk 9 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv]: Done

File: configuration/concepts/lab_tests.csv (hunk 8 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv]: It was part of the serology tests you earlier told me to add to blood sample. Let me delete it.

File: configuration/concepts/lab_tests.csv [hunk 10]
```diff
@@ -95,27 +100,76 @@ d3966267-5882-437b-ae14-18dee9657d14,,,RBCs,Hematíes,Hematíes,RBCs,,LabTest,Co
...
-74ab27c4-c745-45ff-b7e6-e8dbaf933458,,,Trichomonas,Trichomonas,Trichomonas,Trichomonas,,LabTest,Coded,,Positive; Negative;,,,,,,,,
+4bb7df94-7b92-4e5f-8221-064a76d65e17,,,Yeast,Hongos,Hongos,Yeast,,LabTest,Coded,,Positive; Negative,,,,,,,,
+b2a75309-4eb5-414d-9f29-8ecb4ac7dd32,,,Bacterial vaginosis,Vaginosis bacteriana,Vaginosis bacteriana,Bacterial vaginosis,,LabTest,Coded,,Positive; Negative,,,,,,,,
+74ab27c4-c745-45ff-b7e6-e8dbaf933458,,,Trichomonas,Trichomonas,Trichomonas,Trichomonas,,LabTest,Coded,,Positive; Negative,,,,,,,,
 14e99cf4-d294-4905-8837-40e704d8a6ad,,,Vaginal Candidiasis,Candidiasis vaginal,Candidiasis vaginal,Vaginal Candidiasis,,LabTest,Text,,,,,,,,,,
-ab6588f6-87ee-45ca-a6c5-dc1337f83469,,,Gonorrhea,Gonorrea,Gonorrea,Gonorrhea,,LabTest,Coded,,Positive; Negative;,,,,,,,,
-637b4f48-8717-45e2-a5f8-96869e3bb768,,,Chlamydia,Clamidia,Clamidia,Chlamydia,,LabTest,Coded,,Positive; Negative;,,,,,,,,
+088253e3-b556-4e82-b646-2bb784a46577,,,Blood HCG,HCG en sangre,HCG en sangre,Blood HCG,,LabTest,Coded,,Positive; Negative,,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: We said we change this to **Blood HCG (Pregnancy)**

File: configuration/concepts/lab_tests.csv [hunk 11]
```diff
@@ -95,27 +100,76 @@ d3966267-5882-437b-ae14-18dee9657d14,,,RBCs,Hematíes,Hematíes,RBCs,,LabTest,Co
...
+cf1cff9a-0df8-4e2d-8abf-f96a3392f8f0,,,Red Blood Cells in Urine,Glóbulos rojos en la orina,Glóbulos rojos en la orina,Red Blood Cells in Urine,,LabTest,Numeric,,,,,cells per field,,,,,
+ef036fb3-6e8c-4c79-9055-54913e4ef70f,,,White Blood Cells in Urine,Glóbulos blancos en la orina,Glóbulos blancos en la orina,White Blood Cells in Urine,,LabTest,Numeric,,,,,cells per field,,,,,
+9a89b6e4-143b-4c03-b929-b085621f75e5,,,Epithelial Tissue,Tejido epitelial,Tejido epitelial,Epithelial Tissue,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+3fed1729-e4e3-434c-946c-58b3f9e8d963,,,Bacterial Cells,Células bacterianas,Células bacterianas,Bacterial Cells,,LabTest,Coded,,Negative; Trace; Small; Moderate; Large,,,,,,,,
+57bf229a-b469-49f1-9d29-3f6e90120b8d,,,Other Test Observations,Otras observaciones de prueba,Otras observaciones de prueba,Other Test Observations,,LabTest,Text,,,,,,,,,,
+666bee99-7881-418d-8ea1-5d2301aa0640,,,Microscopy Set,Juego de microscopía,Juego de microscopía,Microscopy Set,,LabSet,N/A,,,Crystals;Casts;Aspect;Color;Yeast Cells;Bacteria;Mucus;Red Blood Cells in Urine;White Blood Cells in Urine;Epithelial Tissue;Other Test Observations,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: Remove **aspect** and **colour** from this set and add it to **Urinalysis** and endeavours to have them as the first in the list

File: configuration/concepts/lab_tests.csv [hunk 12]
```diff
@@ -95,27 +100,76 @@ d3966267-5882-437b-ae14-18dee9657d14,,,RBCs,Hematíes,Hematíes,RBCs,,LabTest,Co
...
...40,,,Blood,Sangre,Sangre,Blood,,Sample,N/A,,,CBC;Blood Cross Matching;Blood Grouping;Major Crossmatch;Minor Crossmatch;Auto Control;ABO; Prostate Specific Antigen;Lipid Profile; Coagulation Profile; Thyroid Functional Test;Renal Functional Test; Liver Functional Test; Rapid malaria; Rapid HIV;Blood culture; Fasting blood glucose; Random blood glucose;HbA1C; mALB; 1e3e425c-eee2-4d61-8652-dd20239ec159; Hematocrit; Red Blood Cells; White Blood Cells; Platelets; Plateletcrit;Mean Platelet Volume;PDWs;PDWc;P-LCC;P-LCR;  MCH; MCHC; RDWs; RDWc;Covid19 Ag (Immunofluorescence); Reticulocytes; Neutrophil; Lymphocyte; Monocyte; Basophil; Eosinophil; Creatinine; BUN; Urea; Sodium; Potassium; Calcium Test; Bicarbonate; Chloride; Phosphate; Serum magnesium; Uric acid; Total Protein; Total Bilirubin; Direct Bilirubin; albumin; ALT; AST; ALP; Total Cholesterol; HDL; LDL; Triglycerides; PT; aPTT; INR; C-Reactive Protein; Erythrocyte Sedimentation Rate; TSH; T3; T4; Amylase; Lipase; Rheumatoid Factor; Antistreptolysin O; Prolactin; Sex Hormones;Chlamydia;Gonorrhea;Hepatitis B;Hepatitis C;Helicobacter pylori Antigen;Antistreptolysin O;Rapid HIV;C-Reactive Protein; Rapid Plasma Reagin;Blood HCG,,,,,,,
```
Comment (reagan-meant) [inline configuration/concepts/lab_tests.csv]: Add **Rhesus Factor** to the blood set and add it immediately next to **ABO**

**Review by reagan-meant — CHANGES_REQUESTED** [review]

File: configuration/concepts/lab_tests.csv (hunk 10 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv]: Done

File: configuration/concepts/lab_tests.csv (hunk 11 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv]: Done

File: configuration/concepts/lab_tests.csv (hunk 12 above)
Comment (basrem) [reply configuration/concepts/lab_tests.csv]: Done
